RFC-011-SWAP-Omni-Layer-basic.md

```diff
@@ -0,0 +1,320 @@
+
+- It MUST be a `valid` Omni Layer transaction.
+- It MUST be simple send Omni Layer transaction Class C.
+- The simple send `amount` MUST be equal to the `quantity` parameter in the Omni Layer asset header.
+- It MUST have an output locked with either
+  - a Pay-To-Witness-Script-Hash (P2WSH) `scriptPubKey` derived from `contract_script` (See [BIP141](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#specification) for how to construct the `scriptPubkey` from the `contract_script`).
+  - OR a Pay-To-Witness-Script-Hash nested in Pay-To-Script-Hash (P2WSH(P2SH)) `scriptPubKey` derived from `contract_script` (See [BIP141](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#p2wsh-nested-in-bip16-p2sh) for how to construct the `scriptPubkey` from the `contract_script`).
+- The simple send `recipient` MUST be confirmed as a correct HTLC address, as defined in [HTLC Address](#htlc-address).
+
+
+Both parties MAY watch the blockchain for the deployment transaction, they MUST consider both (1) and (2) addresses.
+
+Such party MUST check the validity of the transaction against Omni Layer Spec.
+Not all valid Bitcoin transactions containing Omni Data are valid Omni Layer transactions.
+
+For example, the following RPC call to omnicore can be used to verify the validity of a deployment transaction:
+- `txid`: the transaction id of the transaction (e.g. `72abc494e877d7e2491ffa6fea3694648aa75e6166e42d6381b04d41ca933fb5`)
+- `referenceaddress`: the HTLC address (e.g. `2N4CHfBpitHNXj9nBcJCFs7BXUaoNVTegBZ`)
+- `valid`: MUST be `true`
+- `amount`: the quantity of asset (e.g. `5300`)
+
+ ```
+$ omnicore-cli "omni_gettransaction" "72abc494e877d7e2491ffa6fea3694648aa75e6166e42d6381b04d41ca933fb5"  
+{
+  "txid": "72abc494e877d7e2491ffa6fea3694648aa75e6166e42d6381b04d41ca933fb5",
+  "fee": "0.00003046",
+  "sendingaddress": "mxuetdW3Yq8KEsvB4L37Z5jmMQMTxZF9Be",
+  "referenceaddress": "2N4CHfBpitHNXj9nBcJCFs7BXUaoNVTegBZ",
+  "ismine": false,
+  "version": 0,
+  "type_int": 0,
+  "type": "Simple Send",
+  "propertyid": 2147483652,
+  "divisible": false,
+  "amount": "5300",
+  "valid": true,
+  "blockhash": "6e9976c5d6c2b4be2928972c93564a0fc0ccdabe4ac581378fad65753559d0b0",
+  "blocktime": 1556091362,
+  "positioninblock": 1,
+  "block": 441,
+  "confirmations": 6
+}
+```
+
+### Redeem transaction
+
+To redeem the HTLC, the redeemer MUST submit a valid Omni Layer transaction transferring all the omni tokens from the HTLC address to their desired address using simple send.
+The transaction should be constructed according to the [Constructing an Omni Layer Transaction](#constructing-an-omni-layer-transaction) section.
+To construct the witness data for the P2WSH output, the redeemer MUST use the procedure described in the [Redeem section of RFC005](./RFC-005-SWAP-Bitcoin-basic.md#redeem).
+
+Both parties SHOULD watch the blockchain for the a bitcoin transaction, that spends the HTLC output and is a valid Omni Layer transaction.
```

Comment:
```suggestion
Both parties SHOULD watch the blockchain for the bitcoin transaction that spends the HTLC output and is a valid Omni Layer transaction.
```

Reply:
Actually the whole sentence is incorrect :)